services/app_service.go

```diff
@@ -49,7 +62,7 @@ func (a *AppService) CreateApp(ctx context.Context, newApp *models.Application,
 		DocumentStatus:  datastore.ActiveDocumentStatus,
 	}
 
-	err := a.appRepo.CreateApplication(ctx, app)
+	err = a.appRepo.CreateApplication(ctx, app)
```

Comment:
Since we already have an index in place, `err` will not be nil for non-unique names. Is it possible to type assert the value of `err` at this stage and return the correct error message back to the client?

Reply:
In hindsight, since we are now calling `IsAppTitleUnique` i will be removing the unique index